collect_app/src/main/java/org/odk/collect/android/backgroundwork/AutoSendTaskSpec.kt

```diff
@@ -44,6 +45,9 @@ class AutoSendTaskSpec : TaskSpec {
     @Inject
     lateinit var instanceAutoSender: InstanceAutoSender
 
+    @Inject
```

Comment:
I don't think there's any reason to inject a creator here instead of just creating the `ProjectSandbox` in `autoSendInstances` like in `FormsUpdater`. Right now, I'd still prefer the `projectId` being the argument to `autoSendInstances`. Maybe I'm missing something though. To be explicit I was thinking:

```kotlin
fun autoSendInstances(projectId: String): Boolean {
   val sandbox = getProjectSandbox(projectId)
   ...
```

Reply:
I know what you meant. It was just playing with other ideas and it wasn't even finished but to be honest I kind of liked this direction.
Without such a creator I would need to inject like 4 new objects into `AutoSendTaskSpec` to construct `ProjectSandbox` there so ok I would make `InstanceAutoSender` a bit smaller but I would clutter `AutoSendTaskSpec` instead.
Maybe having such a creator for `ProjectSandbox` would make sense?

Reply:
Ok I renamed the classes according to your advice and also used in `FormsUpdater`. The only difference is that I passed `ProjectDependencyProviderFactory` via constructor to `FormsUpdater` since `FormsUpdater` is created using dagger it's easier I think. Then in methods were we already passed projectId I just create `ProjectDependencyProvider` using that id.